Dockerfile

```diff
@@ -13,23 +13,23 @@ WORKDIR /workspace
 # Copy the Go Modules manifests
 COPY go.mod go.mod
 COPY go.sum go.sum
-# cache deps before building and copying source so that we don't need to re-download as much
-# and so that source changes don't invalidate our downloaded layer
-RUN go mod download
 
 # Copy the go source
+COPY vendor/ vendor/
+COPY version/ version/
 COPY main.go main.go
+COPY hack/ hack/
 COPY api/ api/
 COPY controllers/ controllers/
 COPY pkg/ pkg/
 COPY install/ install/
 # Build
```

Comment:
IIRC you are missing `git` which should be installed in line 3 and copied right before building the binary.
```
# for getting version info
COPY .git/ .git/
```